nixos/modules/services/web-apps/archtika.nix

```diff
@@ -0,0 +1,312 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.archtika;
+  baseHardenedSystemdOptions = {
+    CapabilityBoundingSet = "";
+    LockPersonality = true;
+    NoNewPrivileges = true;
+    PrivateDevices = true;
+    PrivateTmp = true;
+    ProtectClock = true;
+    ProtectControlGroups = true;
+    ProtectHome = true;
+    ProtectHostname = true;
+    ProtectKernelLogs = true;
+    ProtectKernelModules = true;
+    ProtectKernelTunables = true;
+    ProtectSystem = "strict";
+    RemoveIPC = true;
+    RestrictNamespaces = true;
+    RestrictRealtime = true;
+    RestrictSUIDSGID = true;
+    SystemCallArchitectures = "native";
+    SystemCallFilter = [
+      "@system-service"
+      "~@privileged"
+      "~@resources"
+    ];
+
+    ReadWritePaths = [ "/var/www/archtika-websites" ];
+  };
+in
+{
+  options.services.archtika = {
+    enable = mkEnableOption "archtika service";
+
+    package = mkPackageOption pkgs "archtika" { };
+
+    user = mkOption {
+      type = types.str;
+      default = "archtika";
+      description = "User account under which archtika runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "archtika";
+      description = "Group under which archtika runs.";
+    };
+
+    databaseName = mkOption {
+      type = types.str;
+      default = "archtika";
+      description = "Name of the PostgreSQL database for archtika.";
+    };
+
+    apiPort = mkOption {
+      type = types.port;
+      default = 5000;
+      description = "Port on which the API runs.";
+    };
+
+    apiAdminPort = mkOption {
+      type = types.port;
+      default = 7500;
+      description = "Port on which the API admin server runs.";
+    };
+
+    webAppPort = mkOption {
+      type = types.port;
+      default = 10000;
+      description = "Port on which the web application runs.";
+    };
+
+    domain = mkOption {
+      type = types.str;
+      default = null;
+      description = "Domain to use for the application.";
+    };
+
+    acmeEmail = mkOption {
+      type = types.str;
+      default = null;
+      description = "Email to notify for the SSL certificate renewal process.";
+    };
+
+    dnsProvider = mkOption {
+      type = types.str;
+      default = null;
+      description = "DNS provider for the DNS-01 challenge (required for wildcard domains).";
+    };
+
+    dnsEnvironmentFile = mkOption {
+      type = types.path;
+      default = null;
```

Comment:
Ditto. The `default` value for an option should always concur with the option's `type`, otherwise all NixOS users will need to change this option for their system to build. (Even if they don't use Archtika)